src/org/ods/component/ScanWithTrivyStage.groovy

```diff
@@ -0,0 +1,195 @@
+package org.ods.component
+
+import groovy.transform.TypeChecked
+import groovy.transform.TypeCheckingMode
+import org.ods.services.TrivyService
+import org.ods.services.BitbucketService
+import org.ods.services.NexusService
+import org.ods.util.ILogger
+
+@TypeChecked
+class ScanWithTrivyStage extends Stage {
+
+    static final String STAGE_NAME = 'Trivy Scan'
+    static final String BITBUCKET_TRIVY_REPORT_KEY = "org.opendevstack.trivy"
+    private final TrivyService trivy
+    private final BitbucketService bitbucket
+    private final NexusService nexus
+    private final ScanWithTrivyOptions options
+
+    @SuppressWarnings('ParameterCount')
+    @TypeChecked(TypeCheckingMode.SKIP)
+    ScanWithTrivyStage(def script, IContext context, Map config, TrivyService trivy, BitbucketService bitbucket,
+                      NexusService nexusService, ILogger logger) {
+        super(script, context, logger)
+        if (!config.resourceName) {
+            config.resourceName = context.componentId
+        }
+        if (!config.format) {
+            config.format = 'cyclonedx'
+        }
+        if (!config.scanners) {
+            config.scanners = 'vuln,config,secret,license'
+        }
+        if (!config.vulType) {
+            config.vulType = 'os,library'
+        }
+        if (!config.additionalFlags) {
+            config.additionalFlags = []
+        }
+        // make this param not configurable by user ?
+        if (!config.nexusRepository) {
+            config.nexusRepository = 'leva-documentation'
+        }
+        this.options = new ScanWithTrivyOptions(config)
+        this.trivy = trivy
+        this.bitbucket = bitbucket
+        this.nexus = nexusService
+    }
+
+    protected run() {
+        String errorMessages = ''
+        String reportFile = "trivy-sbom.json"
+        int returnCode = scanViaCli(options.scanners, options.vulType, options.format,
+            options.additionalFlags, reportFile)
+        if ([TrivyService.TRIVY_SUCCESS].contains(returnCode)) {
+            try {
+                URI reportUriNexus = archiveReportInNexus(reportFile, options.nexusRepository)
+                createBitbucketCodeInsightReport(options.nexusRepository ? reportUriNexus.toString() : null,
+                    returnCode, errorMessages)
+                archiveReportInJenkins(!context.triggeredByOrchestrationPipeline, reportFile)
+            } catch (err) {
+                logger.warn("Error archiving the Trivy reports due to: ${err}")
+                errorMessages += "<li>Error archiving Trivy reports</li>"
```

Comment:
This message never is being used in anyplace

Reply:
I have removed said line because as you said it will not be used